packages/design-system/src/components/analytics/AnalyticsContent.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect, useRef } from 'react';
+
+export interface RenderToStringProps {
+  children: string | React.ReactNode;
+  onContentStringRendered: (stringContent: string) => any;
+}
+
+/**
+ * For analytics, we often need to send a string version of the content that the user
+ * is being shown. However, most of our content props allow for React nodes, which
+ * must be rendered before it can be turned into a string. While there is an existing
+ * function in `react-dom/server` that will render JSX components to strings, using
+ * it here would add about 20KB to the bundle and would include html tags, which we
+ * don't care about. This component renders content and sends a string back to the
+ * parent through a callback.
+ */
+export const AnalyticsContent = ({ children, onContentStringRendered }: RenderToStringProps) => {
+  const ref = useRef<HTMLSpanElement>();
+
+  useEffect(() => {
+    if (ref.current) {
+      onContentStringRendered(ref.current.textContent);
+    }
+  });
```

Comment:
[question] would you want to have a conditional variable here as the second param of `useEffect`? Or is the intention that this effect will run on every re-render?

Reply:
I thought about that but didn't know the answer off the top of my head. I figured that was a detail I could figure out if we actually wanted to implement it. What do you think of this option?